File: sdk/vbnet/CardAuthSDK/Models/ApiModels.vb
Imports System
Imports System.Collections.Generic

Namespace CardAuthSDK.Models
    Public Class ApiResponse(Of T)
        Public Property Code As Integer
        Public Property Message As String
        Public Property Data As T
        Public Property Timestamp As Long

        Public ReadOnly Property IsSuccess As Boolean
            Get
                Return Code = 0 OrElse Code = 200
            End Get
        End Property
    End Class

    Public Class CardVerifyResult
        Public Property CardId As Integer
        Public Property CardType As Integer
        Public Property CardTypeText As String
        Public Property Status As Integer
        Public Property StatusText As String
        Public Property ExpireTime As String
        Public Property RemainingDuration As Integer
        Public Property BindDeviceCount As Integer
        Public Property BindLimit As Integer
        Public Property IsPermanent As Boolean
        Public Property IsSoftExpired As Boolean
    End Class

    Public Class CardQueryResult
        Public Property CardType As Integer
        Public Property CardTypeText As String
        Public Property Status As Integer
        Public Property StatusText As String
        Public Property ExpireTime As String
        Public Property RemainingDuration As Integer
        Public Property BindDeviceCount As Integer
        Public Property BindLimit As Integer
        Public Property IsPermanent As Boolean
        Public Property ActivateTime As String
    End Class

    Public Class OnlineCountResult
        Public Property OnlineCount As Integer
        Public Property AppId As Integer
    End Class

    Public Class AnnouncementResult
        Public Property Title As String
        Public Property Content As String
        Public Property Enabled As Boolean
        Public Property Variables As List(Of String)
    End Class
End Namespace

File: sdk/csharp/CardAuthSDK/Models/ApiModels.cs
using System;
using System.Collections.Generic;

namespace CardAuthSDK.Models
{
    public class ApiResponse<T>
    {
        public int Code { get; set; }
        public string Message { get; set; }
        public T Data { get; set; }
        public long Timestamp { get; set; }

        public bool IsSuccess
        {
            get { return Code == 0 || Code == 200; }
        }
    }

    public class CardVerifyResult
    {
        public int CardId { get; set; }
        public int CardType { get; set; }
        public string CardTypeText { get; set; }
        public int Status { get; set; }
        public string StatusText { get; set; }
        public string ExpireTime { get; set; }
        public int RemainingDuration { get; set; }
        public int BindDeviceCount { get; set; }
        public int BindLimit { get; set; }
        public bool IsPermanent { get; set; }
        public bool IsSoftExpired { get; set; }
    }

    public class CardQueryResult
    {
        public int CardType { get; set; }
        public string CardTypeText { get; set; }
        public int Status { get; set; }
        public string StatusText { get; set; }
        public string ExpireTime { get; set; }
        public int RemainingDuration { get; set; }
        public int BindDeviceCount { get; set; }
        public int BindLimit { get; set; }
        public bool IsPermanent { get; set; }
        public string ActivateTime { get; set; }
    }

    public class OnlineCountResult
    {
        public int OnlineCount { get; set; }
        public int AppId { get; set; }
    }

    public class AnnouncementResult
    {
        public string Title { get; set; }
        public string Content { get; set; }
        public bool Enabled { get; set; }
        public List<string> Variables { get; set; }
    }
}
